### PR TITLE
Hotfix/resolve atf add sub menu known issues

### LINKED ISSUE #479: AddCommand API responses GENERIC_ERROR response after Sending  200 addCommand
We are trying to test with 200 AddCommand.

We noticed,
1. for first140 requests SDL core responses SUCCESS
2. remaining 60 Requests it reponse GENERIC_ERROR. but it added to command list and shows in HMI.
Note: We are using default time out 20 sec.

Please let us know if we are doing anything wrong.

Attaching logs.
[SmartDeviceLinkCore-2.txt](https://github.com/smartdevicelink/sdl_core/files/213913/SmartDeviceLinkCore-2.txt)

### PR DESCRIPTION
**Pull request by BrandonHe**
Replace PR #660, to resolve #479 and #621.
Please help to review @jacobkeeler @LuxoftAKutsan @AGaliuzov @madhuy 

According to requirements, change PR name to  **hotfix** and branched off **master**, have rebased and squashed the original commits. 
For detail information, please reference PR #660.

### CODE REVIEW

File: src/components/application_manager/src/request_controller.cc [hunk 1]
```diff
@@ -52,14 +52,19 @@ RequestController::RequestController(const RequestControlerSettings& settings)
     , timer_("AM RequestCtrlTimer",
              new timer::TimerTaskImpl<RequestController>(
                  this, &RequestController::onTimer))
+    , timer_stop_flag_(false)
     , is_low_voltage_(false)
     , settings_(settings) {
   LOG4CXX_AUTO_TRACE(logger_);
   InitializeThreadpool();
+  timer_.Start(0, timer::kSingleShot);
```
Comment (madhuy) [inline src/components/application_manager/src/request_controller.cc]: is it Compiling with 4.1? 'kSingleShot' is not defined I guess.

Comment (BrandonHe) [inline src/components/application_manager/src/request_controller.cc]: Thanks, done with commit [4722196](https://github.com/smartdevicelink/sdl_core/pull/909/commits/47221968bfc52c0d2e8924aee700f11d3406da38)

File: src/components/smart_objects/src/smart_object.cc [hunk 2]
```diff
@@ -733,15 +733,19 @@ void SmartObject::cleanup_data() {
...
       delete m_data.map_value;
+      m_data.map_value = NULL;
       break;
     case SmartType_Array:
       delete m_data.array_value;
+      m_data.array_value = NULL;
       break;
     case SmartType_Binary:
       delete m_data.binary_value;
+      m_data.array_value = NULL;
```
Comment (jacobkeeler) [inline src/components/smart_objects/src/smart_object.cc]: @BrandonHe Also, can these changes be removed, since they are not relevant to the rest of the PR

**Comment by JackLivio** [comment]
@madhuy Please confirm updated PR fixes issues #479 and #621

File: src/components/smart_objects/src/smart_object.cc (hunk 2 above)
Comment (BrandonHe) [inline src/components/smart_objects/src/smart_object.cc]: @jacobkeeler Do I need to send another PR for these small changes? I think these changes is necessary in case of dangling pointer.

**Comment by madhuy** [comment]
@JackLivio Verified  #479 and #621 both are working .

File: src/components/smart_objects/src/smart_object.cc (hunk 2 above)
Comment (BrandonHe) [inline src/components/smart_objects/src/smart_object.cc]: @jacobkeeler OK, thanks!
